### PR TITLE
Amanda's solution

### PR DESCRIPTION
**Pull request by araskog**
This is my first Tjejer Kodar coding challenge. Would love your feedback!

### CODE REVIEW

File: huxflux.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  };
+
+  // Generate a list with a number of random integers between min and max (inclusive), either unordered or orderd (order = false/true)
+  const genRandomInt = (num, min, max, order) => {
+    do {
+      const heltal = randomInt(min, max);
+      list.add(heltal);
+    } while (list.size < num);
+
+    return order ? [...list].sort((a, b) => a - b) : [...list];
```
Comment (rebkova) [inline huxflux.js:20]: Very smooth conversion of an object to an array and use of ternary operators to make this line of code more readable. Big 👍!

File: huxflux.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+// Huxflux returns an array of 7 unique integers
+const huxflux = () => {
```
Comment (rebkova) [inline huxflux.js:2]: Nice that you converted the function to ES6 syntax :)

File: huxflux.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    do {
+      const heltal = randomInt(min, max);
+      list.add(heltal);
+    } while (list.size < num);
+
+    return order ? [...list].sort((a, b) => a - b) : [...list];
+  };
+
+  return genRandomInt(7, 1, 35, false);
+};
```
Comment (rebkova) [inline huxflux.js:24]: I really like how you broke the code into smaller functions and commented when needed. It makes it very readable and understandable. It is an important skill in development, as your code will be read by other developers. Keep it up!

File: index.html [hunk 4]
```diff
@@ -23,19 +23,24 @@
             </div>
             <div class="ball-row flex-row flex-center"></div>
             <div class="huxflux-row flex-row flex-center">
-                <button id="huxflux-button">Huxflux</button>
-                <button id="test-button">Testa din Huxflux</button>
+                <button id="huxflux-button">Prova funktionen</button>
+                <button id="order-button">Prova sortering</button>
```
Comment (rebkova) [inline index.html:27]: I'm happy to see you've stretched the goal and added a sort button as well!

File: index.html [hunk 5]
```diff
@@ -23,19 +23,24 @@
...
                 </ul>
             </div>
-        </div
+        </div>
     </div>
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/gsap/3.5.1/gsap.min.js"></script>
     <script type="module" src="javascripts/ball-creator.js"></script>
-    <script type="module" src="test/test.js"></script>
+    <script type="module" src="javascripts/test.js"></script>
+    <script type="huxflux.js"></script>
```
Comment (rebkova) [inline index.html:42]: Is this line neccessary? + Can one assign a JS file to `type` attribute? Or did you mean `src="huxflux.js"` ?

File: huxflux.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  const randomInt = (min, max) => {
+    min = Math.ceil(min);
+    max = Math.floor(max);
+    return Math.floor(Math.random() * (max - min + 1)) + min;
+  };
+
+  // Generate a list with a number of random integers between min and max (inclusive), either unordered or orderd (order = false/true)
+  const genRandomInt = (num, min, max, order) => {
+    do {
+      const heltal = randomInt(min, max);
```
Comment (rebkova) [inline huxflux.js:16]: For consistency sake, perhaps keep your variable names in English as well?

File: javascripts/ball-creator.js [hunk 7]
```diff
@@ -0,0 +1,72 @@
+// Den här filen skapar bollarna. Du behöver
+// inte ändra nåt här, men tjuvkika gärna!
+
+import { huxflux } from "../huxflux.js";
+let called = 0;
+
+const createBall = (number) => {
```
Comment (rebkova) [inline javascripts/ball-creator.js:7]: Great job using arrow function syntax here as well. In cases where you only pass in one argument (`number` in you case), you can omit the brackets:  
```
const createBall = number => { 
//code to be executed
}
```

File: javascripts/ball-creator.js [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+};
+
+document
+  .querySelector("#huxflux-button")
+  .addEventListener("click", function () {
+    appendBalls(huxflux());
+  });
+
+// Added for sorted balls
+document.querySelector("#order-button").addEventListener("click", function () {
```
Comment (rebkova) [inline javascripts/ball-creator.js:53]: To stick to arrow function synthax, you could write this anonymous function as: 
```
.addEventListener("click", () => {
//code to be executed
}
```

**Review by rebkova — APPROVED** [review]
Well done, Amanda. I am very impressed by your solution and your way of thinking. I've left some minor comments. Keep up the great job!

**Comment by araskog** [comment]
Thanks very much for your feedback @rebkova! I really appreciate it, I learned a lot!

**Comment by rebkova** [comment]
> Thanks very much for your feedback @rebkova! I really appreciate it, I learned a lot!

My pleasure, hope to see you at the next challenge!

**Pull request closed without merging**